src/bp/core/services/workspace-service.ts

```diff
@@ -1,17 +1,27 @@
-import { Logger } from 'botpress/sdk'
-import { defaultPipelines, defaultWorkspace } from 'common/defaults'
+import { AddWorkspaceUserOptions, Logger, RolloutStrategy, WorkspaceRollout } from 'botpress/sdk'
+import { CHAT_USER_ROLE, defaultPipelines, defaultWorkspace } from 'common/defaults'
 import { AuthRole, CreateWorkspace, Pipeline, Workspace, WorkspaceUser } from 'common/typings'
+import { WorkspaceInviteCode, WorkspaceInviteCodesRepository } from 'core/repositories'
 import { StrategyUsersRepository } from 'core/repositories/strategy_users'
 import { WorkspaceUserAttributes, WorkspaceUsersRepository } from 'core/repositories/workspace_users'
-import { ConflictError, NotFoundError } from 'core/routers/errors'
+import { BadRequestError, ConflictError, NotFoundError } from 'core/routers/errors'
 import { inject, injectable, tagged } from 'inversify'
 import _ from 'lodash'
+import nanoid from 'nanoid/generate'
 
 import { TYPES } from '../types'
 
 import { InvalidOperationError } from './auth/errors'
 import { GhostService } from './ghost/service'
 
+export const ROLLOUT_STRATEGIES: RolloutStrategy[] = [
+  'anonymous',
```

Comment:
I guess it's really a matter of preference ! I kind of like anonymous because it fits well with other options (authenticated, authorized), (... and it was in the specs )